programs/web-server-exporter/clickhouse-web-server-exporter.cpp

```diff
@@ -0,0 +1,122 @@
+#include <Common/Exception.h>
+#include <Common/TerminalSize.h>
+
+#include <IO/ReadHelpers.h>
+#include <IO/ReadBufferFromFile.h>
+#include <IO/WriteHelpers.h>
+#include <IO/WriteBufferFromHTTP.h>
+#include <IO/copyData.h>
+#include <IO/createReadBufferFromFileBase.h>
+
+#include <boost/program_options.hpp>
+#include <re2/re2.h>
+#include <filesystem>
+
+namespace fs = std::filesystem;
+
+#define UUID_PATTERN "[\\w]{8}-[\\w]{4}-[\\w]{4}-[\\w]{4}-[\\w]{12}"
+#define EXTRACT_UUID_PATTERN fmt::format(".*/({})/.*", UUID_PATTERN)
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+
+void processTableFiles(const String & url, const fs::path & path, const String & files_prefix, String uuid)
```

Comment:
I think it would be better to check if the file we want to upload already exists on the server. Now if I try to run command like 
```bash
clickhouse-web-server-exporter --files-prefix data --url http://test.com --metadata-path /home/avogar/tmp/server/store/0f2/0f294285-c5e5-4146-8f29-4285c5e5e146/ 
```
 twice I get an error:
```
libc++abi: terminating with uncaught exception of type DB::Exception: Received error from remote server /data-0f294285-c5e5-4146-8f29-4285c5e5e146-all_1_1_0-checksums.txt. HTTP status code: 204 No Content, body: 
Aborted
```